src/providers/journal/journal.ts

```diff
@@ -47,13 +54,38 @@ export class JournalProvider {
 
   getAndConvertOfflineJournal = (): Promise<ExaminerWorkSchedule> =>
     this.dataStore.getItem('JOURNAL')
-      .then(data => JSON.parse(data))
+      .then((data) => {
+        const journalCache: JournalCache = JSON.parse(data);
+        const cachedDate = DateTime.at(journalCache.dateStored);
+        if (this.isCacheTooOld(cachedDate, new DateTime())) {
+          return this.emptyCachedData();
+        }
+        return journalCache.data;
+      })
       .catch(error => error)
 
   saveJournalForOffline = (journalData: ExaminerWorkSchedule) => {
     if (this.networkStateProvider.getNetworkState() === ConnectionStatus.ONLINE) {
-      this.dataStore.setItem('JOURNAL', JSON.stringify(journalData)).then((response) => {});
+      const journalDataToStore: JournalCache = {
+        dateStored: DateTime.now().format('YYYY/MM/DD'),
+        data: journalData,
+      };
+      this.dataStore.setItem('JOURNAL', JSON.stringify(journalDataToStore)).then((response) => {});
     }
   }
 
+  isCacheTooOld = (dateStored: DateTime, now: DateTime):boolean => {
+    return dateStored.daysDiff(now) > this.appConfigProvider.getAppConfig().daysToCacheJournalData;
+  }
+
+  emptyCachedData = () => {
```

Comment:
Ideally we should have a `journal.offline.effects.ts` where we house all of these dataStore interactions in. Much like how we do with the logs. In the absence of that, could we just add some docblocs to outline the purpose of these methods as related to the offline secure storage feature?

Reply:
doc blocks added